### DISCUSSION TITLE
How to use the ReplicaSet retained by argo?

### DISCUSSION THREAD

**Discussion by davidshen84** [argoproj/argo-cd#12763] [Q&A] [answered]
Hi,

When creating an application, argo retains a history of deployments, so when you rollout a new deployment, the old replica set is scaled down to 0 pod.

But from the UI (v2.5.2), there's nothing I can do with those RS, except deleting them. So, my question is, why do we want to keep that RS history? What action/function that actually utilize those retained RS objects?

Thanks.

**Comment by jannfis** [comment, marked as answer]
Hi. This is actually a feature of Kubernetes, and is controlled by a Deployment's [.spec.revisionHistoryLimit](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#revision-history-limit) setting. From a plain Kubernetes point of view, the old ReplicaSets can be used to rollback to previous versions of a pod.

Argo CD only makes this visible, but is not the one accountable for keeping the ReplicaSets around.

**Reply by leaddevops** [reply]
thats rite, but is there a way ArgoCD show only the active replicaset which has replicas running and ignore the rest in UI

**Comment by LosD** [comment]
@jannfis I realize this is old, but I guess ArgoCD doesn't use the old ReplicaSets in its own history and rollback system? If not, would there be any drawback to setting revisionHistoryLimit to 0?
